### PR TITLE
Add sanity check on new batch input types. Add model generation and test.

### PR DESCRIPTION
**Pull request by GuanLuo**

### CODE REVIEW

**Review by Tabrizian — COMMENTED** [review]
Do we need to update the ragged batching docs too https://github.com/triton-inference-server/server/blob/main/docs/ragged_batching.md?

File: qa/L0_batch_input/batch_input_test.py [hunk 1]
```diff
@@ -244,6 +244,93 @@ def test_max_element_count_as_shape(self):
...
 
+    def test_batch_item_shape_flatten(self):
+        # Use 4 set of inputs with shape
+        # [1, 4, 1], [1, 1, 2], [1, 1, 2], [1, 2, 2]
+        self.inputs = []
+        for value in [[1, 4, 1], [1, 1, 2], [1, 1, 2], [1, 2, 2]]:
+            self.inputs.append(
+                [tritonhttpclient.InferInput('RAGGED_INPUT', value, "FP32")])
+            self.inputs[-1][0].set_data_from_numpy(
+                np.full(value, value[0], np.float32))
```
Comment (CoderHam) [inline qa/L0_batch_input/batch_input_test.py:255]: Why are we using np.full instead of np.ones?

File: qa/L0_batch_input/batch_input_test.py [hunk 2]
```diff
@@ -244,6 +244,93 @@ def test_max_element_count_as_shape(self):
...
+            self.assertTrue(False, "unexpected error {}".format(ex))
+
+    def test_batch_item_shape(self):
+        # Use 3 set of inputs with shape [2, 1, 2], [1, 1, 2], [1, 2, 2]
+        inputs = []
+        for value in [[2, 1, 2], [1, 1, 2], [1, 2, 2]]:
+            inputs.append(
+                [tritonhttpclient.InferInput('RAGGED_INPUT', value, "FP32")])
+            inputs[-1][0].set_data_from_numpy(
+                np.full(value, value[0], np.float32))
```
Comment (CoderHam) [inline qa/L0_batch_input/batch_input_test.py:295]: Same here. Does the value of the inputs matter?

**Comment by GuanLuo** [comment]
The ragged batching docs doesn't describe all batch input types in detail, currently it points to the model config if the user wants to learn more about it, so I think the current doc is still accurate.

**Review by CoderHam — CHANGES_REQUESTED** [review]

File: qa/L0_batch_input/batch_input_test.py (hunk 2 above)
Comment (CoderHam) [reply qa/L0_batch_input/batch_input_test.py:295]: I'd change the np.full to be np.ones or hardcode the value being filled instead of making it value[0]

**Review by CoderHam — APPROVED** [review]

File: qa/common/gen_qa_ragged_models.py [hunk 3]
```diff
@@ -471,20 +488,236 @@ def create_modelconfig(models_dir, max_batch, model_version, dtype, backend,
...
+
+    backend_spec = '''
+backend: "{}"
+'''.format(backend)
+    if backend == "tensorflow":
+        backend_spec += '''
+platform: "{}_{}"
+'''.format(backend, platform)
+
+    # Use a different model name for the non-batching variant
```
Comment (dyastremsky) [inline qa/common/gen_qa_ragged_models.py:648]: Should this comment be somewhere else? This function seems to be for the batching variant.

Or maybe "a different model name from the non-batching variant"?

**Review by dyastremsky — CHANGES_REQUESTED** [review]

**Review by dyastremsky — APPROVED** [review]

File: qa/L0_batch_input/batch_input_test.py (hunk 2 above)
Comment (GuanLuo) [reply qa/L0_batch_input/batch_input_test.py:295]: What is the reason for this change? If it is to highlight the value is irrelevant, I think the better way is to clarify it with a comment, using hardcode value may give the opposite impression (i.e. a special value is required)

**Review by dyastremsky — DISMISSED** [review]

File: qa/L0_batch_input/batch_input_test.py (hunk 2 above)
Comment (CoderHam) [reply qa/L0_batch_input/batch_input_test.py:299]: Sure, you can use a comment to clarify the same.

**Review by tanmayv25 — DISMISSED** [review]

**Review by dyastremsky — APPROVED** [review]

**Pull request merged**